Update azure git provider to work with self hosted server

b3go opened the pull request:
Applying the same logic as github and gitlab git-provider in the azure provider.

The provider will accept if the ProviderName 'azure' is part of the hostname.

krancour commented:
@b3go unless I am missing something, it seems this only makes the predicate work for self-hosted Azure devops, but nothing appears to be adjusting the client connection URL for self-hosted. It appears it will still, unconditionally, be of the form `https://dev.azure.com/org/<project>/_git/<repo>`.

b3go commented:
@krancour true, the path is the same on self hosted and cloud services. For older instances, which where upgraded from when it was still called team foundation server there seems to sometimes be a base route /tfs, which would be the only difference to the cloud service urls.

On the self hosted server org is called collection but that's about it. API is the same

krancour commented:
> the path is the same on self hosted and cloud services

But hostname/domain is not.

b3go commented:
Well looks like I was pretty naive at that. Should have given it more thought. Thanks for getting me on the right track. I will fix my PR when I am back!

b3go commented:
This bothered me the whole vacation, usually I don't deliver such botched work. I haven't tested it yet but I incorporated the feedback, fixed the provider connection url and signed of the commits

krancour commented:
> But something else: the codecov check fails because the coverage decreases by 0,08% even though it shows 100% coverage for my patch. Any ideas where I can have a look to fix this?

That won't block the PR from being merged.

b3go commented:
I was finally able to test this in my local environment. It won't work unfortunately. I have outlined the reason in this Discussion: https://github.com/akuity/kargo/discussions/4598

The open-pr promotion step needs to clone the git repository, on Azure DevOps Server this currently only works with user:pw and not PAT. The gitprovider uses this go repo for accessing the Azure Devops Api to create PRs: [here](https://github.com/microsoft/azure-devops-go-api). This can only open connections using PATs. So either Kargo won't be able to clone the repo, or create the pr.

As I've stated in the discussion the only way to make this work is to pass the http.extraheader in git clone/push. I would be happy to contribute to that but need some feedback where to start.
